File: 23-merge-k-sorted-lists/merge-k-sorted-lists.java
class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        // Check if the input array is null or empty
        if (lists == null || lists.length == 0) {
            return null;
        }
        boolean allNull = true;
        for (ListNode node : lists) {
            if (node != null) {
                allNull = false;
                break;
            }
        }
        if (allNull) {
            return null;
        }

        ListNode ret = new ListNode();
        ListNode cur = ret;
        while (true) {
            int minIndex = -1;
            int minValue = Integer.MAX_VALUE;
            for (int i = 0; i < lists.length; i++) {
                if (lists[i] != null && lists[i].val < minValue) {
                    minIndex = i;
                    minValue = lists[i].val;
                }
            }
            if (minIndex == -1) {
                break;
            }

            cur.next = lists[minIndex];
            cur = cur.next;

            // Move to the next node in the list that had the smallest value
            lists[minIndex] = lists[minIndex].next;
        }

        return ret.next;
    }
}

File: 23-merge-k-sorted-lists/merge-k-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeKLists(self, lists):
        li=[]
        for i in lists:
            j=i
            while j!=None:
                li.append(j.val)
                j=j.next
        if len(li)==0:
            return 
        li.sort()

        head=ListNode()
        cur=head
        for i in range(len(li)):
            if i==0:
                head.val=li[i]
                head.next=None
            else:
                temp=ListNode(li[i],None)
                cur.next=temp
                cur=cur.next
        return head

        